File: src/Web/ClientApp/src/backendApi/clients/WorkspaceApiClient.ts
import { IUpdateWorkspaceRequest } from "src/backendApi/models/request/workspace/IUpdateWorkspaceRequest";
import { ICreateWorkspaceRequest } from "src/backendApi/models/request/workspace/ICreateWorkspaceRequest";
import { IGetUserWorkspacesResponse } from "src/backendApi/models/response/IGetUserWorkspacesResponse";
import { IWorkspaceApiModel } from "src/backendApi/models/response/IWorkspaceApiModel";
import { BackendRequestSender } from "src/backendApi/senders/BackendRequestSender";
import { IDeleteWorkspaceRequest } from "src/backendApi/models/request/workspace/IDeleteWorkspaceRequest";
import { IGetWorkspaceByIdRequest } from "src/backendApi/models/request/workspace/IGetWorkspaceByIdRequest";
import { IOperationResult } from "src/backendApi/models/response/OperationResult/IOperationResult";

export class WorkspaceApiClient {
    private static readonly apiUrl: string = "/api/WorkspaceApi/";
    private readonly backendRequestSender: BackendRequestSender = new BackendRequestSender();

    public async getUserWorkspaces(): Promise<IOperationResult<IGetUserWorkspacesResponse>> {
        return this.backendRequestSender.sendGetRequest(`${WorkspaceApiClient.apiUrl}GetUserWorkspaces`);
    }

    public createWorkspace(requestModel: ICreateWorkspaceRequest): Promise<IOperationResult<void>> {
        return this.backendRequestSender.sendPostRequest(`${WorkspaceApiClient.apiUrl}Create`, requestModel);
    }

    public deleteWorkspace(requestModel: IDeleteWorkspaceRequest): Promise<IOperationResult<void>> {
        return this.backendRequestSender.sendPostRequest(`${WorkspaceApiClient.apiUrl}Delete`, requestModel);
    }

    public getWorkspaceById(requestModel: IGetWorkspaceByIdRequest): Promise<IOperationResult<IWorkspaceApiModel>> {
        return this.backendRequestSender.sendPostRequest(`${WorkspaceApiClient.apiUrl}GetById`, requestModel);
    }

    public updateWorkspace(requestModel: IUpdateWorkspaceRequest): Promise<IOperationResult<void>> {
        return this.backendRequestSender.sendPostRequest(`${WorkspaceApiClient.apiUrl}Update`, requestModel);
    }
}

File: tests/Web.IntegrationTest/Controllers/WorkspaceApiControllerTests/WorkspaceApiClient.cs
using System;
using System.Net.Http;
using System.Net.Http.Json;
using System.Text.Json;
using System.Threading.Tasks;
using Web.Models.Workspace.Request;
using Web.Models.Workspace.Response;

namespace Web.IntegrationTest.Controllers.WorkspaceApiControllerTests
{
    internal class WorkspaceApiClient
    {
        private const string ApiUrlBase = "api/WorkspaceApi/";

        private readonly JsonSerializerOptions jsonSerializerOptions = new JsonSerializerOptions
        {
            PropertyNameCaseInsensitive = true
        };

        private readonly HttpClient httpClient;

        public WorkspaceApiClient(HttpClient httpClient)
        {
            this.httpClient = httpClient;
        }

        public async Task<HttpResponseMessage> GetUserWorkspacesHttpResponseAsync()
        {
            return await httpClient.SendAsync(new HttpRequestMessage(HttpMethod.Get, $"{ApiUrlBase}GetUserWorkspaces"));
        }

        public async Task<GetUserWorkspacesResponse> GetUserWorkspacesParsedResponseAsync()
        {
            HttpResponseMessage response = await GetUserWorkspacesHttpResponseAsync();
            string responseText = await response.Content.ReadAsStringAsync();
            GetUserWorkspacesResponse responseData = JsonSerializer.Deserialize<GetUserWorkspacesResponse>(responseText, jsonSerializerOptions) ?? throw new NullReferenceException();
            return responseData;
        }

        public async Task<HttpResponseMessage> GetByIdHttpResponseAsync(GetWorkspaceByIdRequest requestModel)
        {
            return await httpClient.SendAsync(new HttpRequestMessage(HttpMethod.Post, $"{ApiUrlBase}GetById")
            {
                Content = JsonContent.Create(requestModel)
            });
        }

        public async Task<WorkspaceApiModel> GetByIdParsedResponseAsync(GetWorkspaceByIdRequest requestModel)
        {
            HttpResponseMessage response = await GetByIdHttpResponseAsync(requestModel);
            string responseText = await response.Content.ReadAsStringAsync();
            WorkspaceApiModel responseData = JsonSerializer.Deserialize<WorkspaceApiModel>(responseText, jsonSerializerOptions) ?? throw new NullReferenceException();
            return responseData;
        }

        public async Task<HttpResponseMessage> CreateAsync(CreateWorkspaceRequest requestModel)
        {
            return await httpClient.SendAsync(new HttpRequestMessage(HttpMethod.Post, $"{ApiUrlBase}Create")
            {
                Content = JsonContent.Create(requestModel)
            });
        }

        public async Task<HttpResponseMessage> UpdateAsync(UpdateWorkspaceRequest requestModel)
        {
            return await httpClient.SendAsync(request: new HttpRequestMessage(HttpMethod.Post, $"{ApiUrlBase}Update")
            {
                Content = JsonContent.Create(requestModel)
            });
        }

        public async Task<HttpResponseMessage> DeleteAsync(DeleteWorkspaceRequest requestModel)
        {
            return await httpClient.SendAsync(new HttpRequestMessage(HttpMethod.Post, $"{ApiUrlBase}Delete")
            {
                Content = JsonContent.Create(requestModel)
            });
        }
    }
}